### PR TITLE
Remove berkeley db

### PR DESCRIPTION
**Pull request by whyamiroot**
Continued #5700. I removed the database switching logic, as suggested by @TheCharlatan, and unrelated changes in the functional test's `CMakeLists.txt`

### CODE REVIEW

File: tests/core_tests/chaingen_main.cpp [hunk 1]
```diff
@@ -106,7 +106,7 @@ int main(int argc, char* argv[])
     GENERATE_AND_PLAY(gen_chain_switch_1);
     GENERATE_AND_PLAY(gen_ring_signature_1);
     GENERATE_AND_PLAY(gen_ring_signature_2);
-    //GENERATE_AND_PLAY(gen_ring_signature_big); // Takes up to XXX hours (if CRYPTONOTE_MINED_MONEY_UNLOCK_WINDOW == 10)
+    GENERATE_AND_PLAY(gen_ring_signature_big); // Takes up to XXX hours (if CRYPTONOTE_MINED_MONEY_UNLOCK_WINDOW == 10)
```
Comment (TheCharlatan) [inline tests/core_tests/chaingen_main.cpp]: Why are you adding this test back?

**Comment by TheCharlatan** [comment]
I'm in favour of squashing this all together to a single commit. You might not be the original author, but since there are hardly any additions involved, I'd be fine with that.

File: tests/core_tests/chaingen_main.cpp (hunk 1 above)
Comment (whyamiroot) [reply tests/core_tests/chaingen_main.cpp]: Oh, sorry, I didn't notice it was commented out before. Is that OK if remove this change inside the a3ed25b commit and force push it?

**Comment by whyamiroot** [comment]
OK, I'll squash it

**Comment by whyamiroot** [comment]
@TheCharlatan Done

File: src/blockchain_db/blockchain_db.cpp [hunk 2]
```diff
@@ -35,17 +35,6 @@
...
-#include "berkeleydb/db_bdb.h"
-#endif
-
-static const char *db_types[] = {
-  "lmdb",
-#ifdef BERKELEY_DB
-  "berkeley",
-#endif
-  NULL
-};
```
Comment (moneromooo-monero) [inline src/blockchain_db/blockchain_db.cpp]: I'd rather the selection mechanism stays. I've said that for quite a while, but since it seems nothing else is being added, maybe that could go. Anyone has an opinion here ?

File: src/blockchain_utilities/blockchain_export.cpp [hunk 3]
```diff
@@ -1,5 +1,4 @@
-// Copyright (c) 2014-2019, The Monero Project
-//
+// Copyright (c) 2014-2019, The Monero Project //
```
Comment (moneromooo-monero) [inline src/blockchain_utilities/blockchain_export.cpp]: Mistake

File: src/blockchain_utilities/blockchain_import.cpp [hunk 4]
```diff
@@ -93,44 +92,6 @@ int get_db_flags_from_mode(const std::string& db_mode)
...
-    {
-      db_mode = db_args[0];
-    }
-  }
-  if (! db_mode.empty())
-  {
-    db_flags = get_db_flags_from_mode(db_mode);
-  }
-  return 0;
-}
```
Comment (moneromooo-monero) [inline src/blockchain_utilities/blockchain_import.cpp]: Why remove this ?

File: src/cryptonote_core/cryptonote_core.cpp [hunk 5]
```diff
@@ -219,6 +219,7 @@ namespace cryptonote
               m_mempool(m_blockchain_storage),
               m_blockchain_storage(m_mempool),
               m_miner(this),
+              m_miner_address(boost::value_initialized<account_public_address>()),
```
Comment (moneromooo-monero) [inline src/cryptonote_core/cryptonote_core.cpp]: This is unrelated

File: src/cryptonote_core/cryptonote_core.h [hunk 6]
```diff
@@ -1014,6 +1014,7 @@ namespace cryptonote
 
      //m_miner and m_miner_addres are probably temporary here
      miner m_miner; //!< miner instance
+     account_public_address m_miner_address; //!< address to mine to (for miner instance)
```
Comment (moneromooo-monero) [inline src/cryptonote_core/cryptonote_core.h]: Remove

File: src/blockchain_utilities/blockchain_import.cpp (hunk 4 above)
Comment (whyamiroot) [reply src/blockchain_utilities/blockchain_import.cpp]: I haven't found any usage of `db_flags` except for printing it to the logs here
https://github.com/monero-project/monero/pull/5878/files/7dc14c71d0e242a3253234113027c383809b5d6c#diff-e130c0e38f072141b5e24d17ec7dfa24L756
As there is no need to parse db type (assuming we remove db switching logic), I figured I could remove this function completely

File: src/blockchain_utilities/blockchain_export.cpp (hunk 3 above)
Comment (whyamiroot) [reply src/blockchain_utilities/blockchain_export.cpp]: Fixed

File: src/cryptonote_core/cryptonote_core.cpp (hunk 5 above)
Comment (whyamiroot) [reply src/cryptonote_core/cryptonote_core.cpp]: Fixed

File: src/cryptonote_core/cryptonote_core.h (hunk 6 above)
Comment (whyamiroot) [reply src/cryptonote_core/cryptonote_core.h]: Fixed

File: src/blockchain_utilities/blockchain_import.cpp (hunk 4 above)
Comment (moneromooo-monero) [reply src/blockchain_utilities/blockchain_import.cpp]: OK, I see this is now handled by core, so this is indeed obsolete.

Comment (moneromooo-monero) [reply src/blockchain_utilities/blockchain_import.cpp]: The general opinion is that it can go.

File: src/blockchain_db/blockchain_db.cpp (hunk 2 above)
Comment (moneromooo-monero) [reply src/blockchain_db/blockchain_db.cpp]: The general opinion is that it can go.
